packages/core/useFetch/index.ts

```diff
@@ -168,6 +185,46 @@ function isFetchOptions(obj: object): obj is UseFetchOptions {
   return containsProp(obj, 'immediate', 'refetch', 'beforeFetch', 'afterFetch')
 }
 
+async function toURLSearchParams(body: any): Promise<URLSearchParams> {
```

Comment:
I'm not sure I understand the purpose of this function. The goal seems to be to deal with the fact that the user may have accidentally passed a search param that is an object, rather than a primitive value type.

1. I'm not sure why it's asynchronous. I don't see any async work being done.
2. It doesn't seem like a good idea to JSON encode _everything_, because then whatever is on the other end of the network will have to JSON decode _each value_. In other words, it will get a bunch of key/value pairs, and then be forced to `JSON.parse` (or the server language equivalent) for every single value in there, even if they're all strings.

I strongly think the onus of doing this work (and deciding if it's even necessary) should be on the developer making the fetch, and we should throw this logic out and let them handle it. They know their server (presumably). And this is covering a use case that generally doesn't exist in the 99% case. Most of the time, if someone is sending search params, they're value-type primitives, not objects. This is penalizing the 99% use case to accommodate a one-off scenario.